src/openrct2/world/Map.cpp

```diff
@@ -1539,6 +1548,104 @@ void map_update_tiles()
     }
 }
 
+void MapInvalidateHeightCache()
+{
+    _mapHighestTileHeightLoopPosition = 0;
+}
+
+void MapUpdateHeightCache(bool fullCheck)
+{
+    constexpr int32_t mapTileCount = MAXIMUM_MAP_SIZE_TECHNICAL * MAXIMUM_MAP_SIZE_TECHNICAL;
+
+    if (fullCheck)
+        _mapHighestTileHeightLoopPosition = 0;
+
+    if (_mapHighestTileHeightLoopPosition == mapTileCount - 1)
```

Comment:
You missed this one

Reply:
![](https://i.kym-cdn.com/entries/icons/original/000/000/554/picard-facepalm.jpg)